### PR TITLE
Adds Windows security documentation

### LINKED ISSUE #1714: [Windows] Security Documentation
Add documentation for the security plugin on a Windows machine

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #1714 

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
Could I have @peternied or anyone from @opensearch-project/security review this PR please?

File: _security-plugin/configuration/security-admin.md
Comment (cwperks) [inline _security-plugin/configuration/security-admin.md:310]: Powershell should work as well

File: _security-plugin/configuration/security-admin.md
Comment (cwperks) [inline _security-plugin/configuration/security-admin.md:325]: _.bat_

File: _security-plugin/configuration/security-admin.md
Comment (cwperks) [inline _security-plugin/configuration/security-admin.md:312]: Is `call` necessary? From what I am reading it looks like `call` was introduced to call batch scripts within a batch script. 

Reference: https://stackoverflow.com/a/14732373

**Review by cwperks — COMMENTED** [review]
@kolchfa-aws Looks good! Can you update from `securityadmin.sh` to `securityadmin.bat` in the windows section? I will approve once the PR is updated.

File: _security-plugin/configuration/security-admin.md
Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md:310]: Powershell has a different method of calling .bat files. I didn't want to make it too complicated, so I assume the users who are familiar with Powershell can translate the commands easily. Any thoughts?

File: _security-plugin/configuration/security-admin.md
Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md:312]: I tried without `call` and it didn't work: it said that .\ is not recognized as a command.

File: _security-plugin/configuration/security-admin.md
Comment (cwperks) [reply _security-plugin/configuration/security-admin.md:310]: I wouldn't assume that. I imagine there are quite a number of people today who are learning how to interact with a terminal for the first time via Powershell.

**Comment by kolchfa-aws** [comment]
@cwperks Thank you for reviewing! I updated to .bat and addressed the comments - please let me know your thoughts.

File: _security-plugin/configuration/security-admin.md
Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md:310]: Sure. I am referencing a thread here https://stackoverflow.com/questions/20645326/safest-way-to-run-bat-file-from-powershell-script. Which method of running the script from Powershell do you think I should include?

File: _security-plugin/configuration/security-admin.md
Comment (cwperks) [reply _security-plugin/configuration/security-admin.md:312]: Interesting. Does it work without the `.\`? i.e. `plugins/opensearch-security/tools/securityadmin.bat` instead of `call ./plugins/opensearch-security/tools/securityadmin.bat`

The official documentation from MS seems to indicate that `call` does not have any effect outside of a script or batch file: https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/call

>  Note

> Call has no effect at the command prompt when it is used outside of a script or batch file.

I bring this up because if we can use the same set of commands on both `cmd` and `Powershell` then it simplifies the documentation.

Comment (cwperks) [reply _security-plugin/configuration/security-admin.md:312]: I just tried with without call on `cmd` on `Windows Server 2016 Datacenter` and it ran the script. 

Running: `.\plugins\opensearch-security\tools\install_demo_configuration.bat` within the OpenSearch home directory.

Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md]: @cwperks thank you for testing and confirming! It must be me then. I updated the docs.

Comment (cwperks) [reply _security-plugin/configuration/security-admin.md]: Last thing to make sure the same command works on both `cmd` and `powershell`. In my testing it looks like `cmd` requires backslashes while powershell can work with either. I think we should make all of these windows commands use backslashes so that it works in both terminals.

Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md]: Interesting. I'm on Windows 10, and both forward slashes and backslashes work in command prompt. In any case, I have updated the docs to have backslashes only. Thank you!

**Review by cwperks — APPROVED** [review]

**Review by cwillum — APPROVED** [review]
LGTM

File: _security-plugin/configuration/index.md
Comment (vagimeli) [inline _security-plugin/configuration/index.md:13]: I recommend breaking this sentence into two sentences: "The plugin includes demo certificates so that you can get up and running quickly. Before using OpenSearch in a production environment, you must configure it manually:"

**Review by vagimeli — APPROVED** [review]
One suggestion. LGTM

File: _security-plugin/configuration/security-admin.md
Comment (natebower) [inline _security-plugin/configuration/security-admin.md:8]: Should the heading and the title match (with vs. using)?

File: _security-plugin/configuration/security-admin.md
Comment (natebower) [inline _security-plugin/configuration/security-admin.md:310]:
```suggestion
When running the example commands in the preceding sections, use the **command prompt** or **Powershell**. Open the command prompt by entering `cmd`, or Powershell by entering `powershell`, in the search box next to **Start** on the taskbar. 
```

File: _security-plugin/configuration/tls.md
Comment (natebower) [inline _security-plugin/configuration/tls.md:94]:
```suggestion
Admin certificates are regular client certificates that have elevated rights to perform administrative tasks. You need an admin certificate to change the security plugin configuration using [`plugins/opensearch-security/tools/securityadmin.sh`]({{site.url}}{{site.baseurl}}/security-plugin/configuration/security-admin/) or the REST API. Admin certificates are configured in `opensearch.yml` by stating their DN(s):
```

File: _security-plugin/configuration/yaml.md
Comment (natebower) [inline _security-plugin/configuration/yaml.md:10]: In "on other clusters", is "on" the right preposition?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and change and let me know if you have any questions. Thanks!

File: _security-plugin/configuration/security-admin.md
Comment (kolchfa-aws) [reply _security-plugin/configuration/security-admin.md:8]: In this situation, it makes perfect sense, so I'll change it. Sometimes though we have to keep them different.

**Pull request merged**
